### DISCUSSION TITLE
采集容器标准输出，ExcludeK8sLabel参数不生效

### DISCUSSION THREAD

**Discussion by hanzoi** [alibaba/ilogtail#501] [Help] [answered]
enable: true
inputs:
  - Type: service_docker_stdout
    Stdout: false
    Stderr: true
    BeginLineRegex: "\\d+-\\d+-\\d+.*"
    ExcludeK8sLabel:
      app: "appadmin"
      app: "appadmin-webpage"
      app: "h2v5"
      app: "etcd"
      app: "redis" 
    ExternalK8sLabelTag: 
      app: service
    K8sNamespaceRegex: dailyyoga

我们想通过这种方式采集go程序打印到 stderr 的 panic 信息，故需要通过 ExcludeK8sLabel 来排除采集 php 容器，但是目前只有在配置 IncludeK8sLabel 并定义全部 go 程序后才可生效，无法通过   ExcludeK8sLabel 排除 php 程序的方式（go程序量很大，php只是少数）

<img width="1153" alt="image" src="https://user-images.githubusercontent.com/61114924/199424604-30606b30-6800-4571-bda6-bcf941f16812.png">

**Comment by messixukejia** [comment, marked as answer]
只有在配置 IncludeK8sLabel 并定义全部 go 程序后才可生效 -- 上面yaml配置，好像跟这句话没什么关系。问题描述不是太清楚，建议补充下更详细的场景。

**Reply by hanzoi** [reply]
简单来说就是配置 ExcludeK8sLabel 不生效，但是配置 IncludeK8sLabel 生效，如上配置 yaml 里面写得这些 label 对应的 pod 日志都被采集了

**Reply by messixukejia** [reply]
key都是app？这种的要用正则配的，看下文档https://github.com/alibaba/ilogtail/blob/d98e0ef98febfacd876217d5df7cbbbd92e0ad84/docs/cn/data-pipeline/input/input-docker-stdout.md

另外，麻烦将解决的问题回馈下社区，通过宣传文章、文档改进、代码等方式都可以。
